opentelemetry-auto-instrumentation/src/opentelemetry/auto_instrumentation/auto_instrumentation.py

```diff
@@ -15,22 +15,37 @@
 # limitations under the License.
 
 from logging import getLogger
-from runpy import run_path
+from os import environ, execl, getcwd
+from os.path import abspath, dirname, pathsep
+from shutil import which
 from sys import argv
 
-from pkg_resources import iter_entry_points
-
 logger = getLogger(__file__)
 
 
 def run() -> None:
 
-    for entry_point in iter_entry_points("opentelemetry_instrumentor"):
-        try:
-            entry_point.load()().instrument()  # type: ignore
-            logger.debug("Instrumented %s", entry_point.name)
+    python_path = environ.get("PYTHONPATH")
+
+    if not python_path:
+        python_path = []
+
+    else:
+        python_path = python_path.split(pathsep)
+
+    cwd_path = getcwd()
+
+    if cwd_path not in python_path:
+        python_path.insert(0, cwd_path)
```

Comment:
I'm curios about what is the case fixed by it?

Reply:
This need was revealed while implementing the Django instrumentation. Django applications are launched with their own executable which expects that the current working directory is part of the settings module path, so that it can be found. Since we are running that executable inside our own executable, the current working directory is changed to the location of our own executable, adding the current working directory to the `PYTHONPATH` fixes this problem.